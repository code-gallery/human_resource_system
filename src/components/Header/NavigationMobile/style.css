.mobile-nav {
  position: fixed;
  top: 50px;
  left: -100%;
  width: 100%;
  height: 100%;
  padding: 20px;
  transition: all .2s linear;
  word-wrap: break-word;
  background-color: #fff;
}

.mobile-nav.hasNotification {
  top: 100px;
}

.mobile-nav .mobile-user-info {
  word-wrap: break-word;
}

.mobile-nav .mobile-user-info:before,
.mobile-nav .mobile-user-info:after {
  display: table;
  content: ' ';
}

.mobile-nav .mobile-user-info:after {
  clear: both;
}

.mobile-nav .Avatar {
  display: inline-block;
  position: relative;
  top: -20px;
}

.mobile-nav .Avatar-icon-danger {
  margin-right: 0;
}

.mobile-nav .mobile-user-info p {
  font-size: 17px;
  overflow: hidden;
  text-align: left;
  color: #44474c;
  display: inline-block;
}

.mobile-nav .mobile-user-info p strong {
  font-family: 'Graphik-Semibold';
  display: block;
}

.mobile-nav .mobile-user-info p a {
  font-size: 17px;
  color: #498df0;
}

.mobile-nav ul {
  margin-top: 20px;
  word-wrap: break-word;
  border-top: solid 1px #e7e5ee;
}

.mobile-nav ul.first {
  margin-top: 0;
}

.mobile-nav ul li {
  margin-top: 20px;
}

.mobile-nav ul li a {
  font-size: 17px;
  color: #44474c;
}

.mobile-nav ul li a:hover {
  text-decoration: underline;
}

.mobile-nav.active {
  left: 0;
}

.mobile-nav .profile-nav-bioID img {
  top: 0;
}
