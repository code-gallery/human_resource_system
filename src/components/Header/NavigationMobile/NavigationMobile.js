import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import classNames from 'classnames'
import { ROUTE_URL } from 'containers/constants.js'
import Avatar from 'components/Avatar'
import Assets from 'components/Profile/ProfileInfo/assets'
import './style.css'

const NavigationMobile = ({
  intelligentLink,
  user,
  active,
  organisations,
  setMobileNavActive,
  hasNotification
}) => {
  const cssNav = classNames('mobile-nav', { active }, { hasNotification })
  const avatarIcon = (user.biometrics_status !== 'complete') ? 'danger' : ''
  const showBiometricLink = user.biometrics_status !== 'complete'

  return (
    <div className={cssNav}>
      <div className="mobile-user-info">
        <Avatar imageUrl={user.profile_image} size="45" icon={avatarIcon} />
        <p>
          <strong>{user.first_name} {user.last_name}</strong>
          {window.location.pathname === ROUTE_URL.profile ? (
            <a onClick={() => setMobileNavActive(false)} className="mobile-profile-link">View Profile</a>
          ) : (
            <Link to={ROUTE_URL.profile}>View Profile</Link>
          )}
        </p>
      </div>
      <ul className="first">
        {showBiometricLink &&
          <li className="profile-nav-bioID">
            {window.location.pathname === ROUTE_URL.helpBiometric ? (
              <span>
                <a onClick={() => setMobileNavActive(false)} className="mobile-profile-link">Identity Not Verified</a>
                <img src={Assets.iconDanger} alt="icon warning" />
              </span>
            ) : (
              <span>
                <Link to={ROUTE_URL.helpBiometric}>Identity Not Verified</Link>
                <img src={Assets.iconDanger} alt="icon warning" />
              </span>
            )}
          </li>
        }
        <li>
          {window.location.pathname === ROUTE_URL.userVerifications ? (
            <a onClick={() => setMobileNavActive(false)} className="mobile-profile-link">Verifications</a>
          ) : (
            <Link to={ROUTE_URL.userVerifications}>My Verifications</Link>
          )}
        </li>
        {
          /* Verifications for Org */
          organisations.map((item, index) => {
            let link = ROUTE_URL.orgVerifications
            link = link.replace(':orgId', item.id)
            return (
              <li key={index}>
                {window.location.pathname === link ? (
                  <a onClick={() => setMobileNavActive(false)} className="h-verifications">{item.name}</a>
                ) : (
                  <Link className="h-verifications" to={link}>{item.name}</Link>
                )}
              </li>
            )
          })
        }
        <li>
          {window.location.pathname === ROUTE_URL.settingsAccount ? (
            <a onClick={() => setMobileNavActive(false)} className="mobile-profile-link">Settings</a>
          ) : (
            <Link to={ROUTE_URL.settingsAccount}>Settings</Link>
          )}
        </li>
        <li><a href={ROUTE_URL.publicSiteHelp}>Help</a></li>
      </ul>
      { organisations && organisations.length > 0 &&
        <ul>
          <li>Organisation Profile</li>
          {
            /* Profile for Org */
            organisations.map((item, index) => {
              let link = ROUTE_URL.orgProfile
              link = link.replace(':orgId', item.id)
              return (
                <li key={index}>
                  {window.location.pathname === link ? (
                    <a onClick={() => setMobileNavActive(false)} className="mobile-profile-link">{item.name}</a>
                  ) : (
                    <Link to={link}>{item.name}</Link>
                  )}
                </li>
              )
            })
          }
        </ul>
      }
      <ul>
        {/* <li>
          {window.location.pathname === ROUTE_URL.people ? (
            <a onClick={() => setMobileNavActive(false)} className="mobile-profile-link">People</a>
          ) : (
            <Link to={ROUTE_URL.people}>People</Link>
          )}
        </li>*/}
        <li> 
          {window.location.pathname === ROUTE_URL.organisations ? (
            <a onClick={() => setMobileNavActive(false)} className="mobile-profile-link">Organisations</a>
          ) : (
            <Link to={ROUTE_URL.organisations}>Organisations</Link>
          )}
        </li>
        {intelligentLink &&
          <li className="hidden-xs"><a href={intelligentLink}>CV Builder</a></li>
        }
      </ul>
      <ul>
        <li><Link to={ROUTE_URL.logout}>Log Out</Link></li>
      </ul>
    </div>
  )
}

NavigationMobile.propTypes = {
  user: PropTypes.object,
  intelligentLink: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.bool
  ]),
  organisations: PropTypes.array.isRequired,
  active: PropTypes.bool.isRequired,
  setMobileNavActive: PropTypes.func.isRequired,
  hasNotification: PropTypes.bool.isRequired
}

export default NavigationMobile
