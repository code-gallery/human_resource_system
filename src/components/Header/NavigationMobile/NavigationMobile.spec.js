import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import NavigationMobile from './NavigationMobile'

describe('components/Header/NavigationMobile', () => {
  const commonProps = {
    user: {
      first_name: 'Jane',
      last_name: 'Doe',
      profile_image: 'https://appii.s3.eu-west-2.amazonaws.com/profile_images/1510235964102.jpg',
      biometrics_status: 'complete'
    },
    active: true,
    organisations: [ { id: 12, name: 'Applied Blockhain Ltd' } ],
    setMobileNavActive: jest.fn(),
    hasNotification: false
  }

  describe('when intelligentLink is valid', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps,
        intelligentLink: 'http://google.co.uk'
      }

      const tree = renderer.create(
        <Router>
          <NavigationMobile {...props} />
        </Router>
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when intelligentLink is false', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps,
        intelligentLink: false
      }

      const tree = renderer.create(
        <Router>
          <NavigationMobile {...props} />
        </Router>
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when user has no organisations', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps,
        organisations: []
      }

      const tree = renderer.create(
        <Router>
          <NavigationMobile {...props} />
        </Router>
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when active is false', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps,
        active: false
      }

      const tree = renderer.create(
        <Router>
          <NavigationMobile {...props} />
        </Router>
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when biometrics_status failed', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps,
        user: {
          ...commonProps.user,
          biometrics_status: 'failed'
        }
      }

      const tree = renderer.create(
        <Router>
          <NavigationMobile {...props} />
        </Router>
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
