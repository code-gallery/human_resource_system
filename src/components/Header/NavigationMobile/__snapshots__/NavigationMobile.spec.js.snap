// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Header/NavigationMobile when active is false renders correctly 1`] = `
<div
  className="mobile-nav"
>
  <div
    className="mobile-user-info"
  >
    <div
      className="Avatar Avatar-white Avatar-size-45"
    >
      <img
        alt="logo"
        src="https://appii.s3.eu-west-2.amazonaws.com/profile_images/1510235964102.jpg"
      />
    </div>
    <p>
      <strong>
        Jane
         
        Doe
      </strong>
      <a
        href="/profile"
        onClick={[Function]}
      >
        View Profile
      </a>
    </p>
  </div>
  <ul
    className="first"
  >
    <li>
      <a
        href="/personal-verification"
        onClick={[Function]}
      >
        My Verifications
      </a>
    </li>
    <li>
      <a
        className="h-verifications"
        href="/organisations/12/verifications"
        onClick={[Function]}
      >
        Applied Blockhain Ltd
      </a>
    </li>
    <li>
      <a
        href="/settings/account"
        onClick={[Function]}
      >
        Settings
      </a>
    </li>
    <li>
      <a
        href="https://appii.io/faqs/"
      >
        Help
      </a>
    </li>
  </ul>
  <ul>
    <li>
      Organisation Profile
    </li>
    <li>
      <a
        href="/organisations/12"
        onClick={[Function]}
      >
        Applied Blockhain Ltd
      </a>
    </li>
  </ul>
  <ul>
    <li>
      <a
        href="/people"
        onClick={[Function]}
      >
        People
      </a>
    </li>
    <li>
      <a
        href="/organisations"
        onClick={[Function]}
      >
        Organisations
      </a>
    </li>
  </ul>
  <ul>
    <li>
      <a
        href="/logout"
        onClick={[Function]}
      >
        Log Out
      </a>
    </li>
  </ul>
</div>
`;

exports[`components/Header/NavigationMobile when biometrics_status failed renders correctly 1`] = `
<div
  className="mobile-nav active"
>
  <div
    className="mobile-user-info"
  >
    <div
      className="Avatar Avatar-white Avatar-size-45"
    >
      <img
        alt="logo"
        src="https://appii.s3.eu-west-2.amazonaws.com/profile_images/1510235964102.jpg"
      />
      <span
        className="Avatar-icon-danger"
      >
        Â 
      </span>
    </div>
    <p>
      <strong>
        Jane
         
        Doe
      </strong>
      <a
        href="/profile"
        onClick={[Function]}
      >
        View Profile
      </a>
    </p>
  </div>
  <ul
    className="first"
  >
    <li
      className="profile-nav-bioID"
    >
      <span>
        <a
          href="/help/biometric"
          onClick={[Function]}
        >
          Identity Not Verified
        </a>
        <img
          alt="icon warning"
          src="icon-danger.png"
        />
      </span>
    </li>
    <li>
      <a
        href="/personal-verification"
        onClick={[Function]}
      >
        My Verifications
      </a>
    </li>
    <li>
      <a
        className="h-verifications"
        href="/organisations/12/verifications"
        onClick={[Function]}
      >
        Applied Blockhain Ltd
      </a>
    </li>
    <li>
      <a
        href="/settings/account"
        onClick={[Function]}
      >
        Settings
      </a>
    </li>
    <li>
      <a
        href="https://appii.io/faqs/"
      >
        Help
      </a>
    </li>
  </ul>
  <ul>
    <li>
      Organisation Profile
    </li>
    <li>
      <a
        href="/organisations/12"
        onClick={[Function]}
      >
        Applied Blockhain Ltd
      </a>
    </li>
  </ul>
  <ul>
    <li>
      <a
        href="/people"
        onClick={[Function]}
      >
        People
      </a>
    </li>
    <li>
      <a
        href="/organisations"
        onClick={[Function]}
      >
        Organisations
      </a>
    </li>
  </ul>
  <ul>
    <li>
      <a
        href="/logout"
        onClick={[Function]}
      >
        Log Out
      </a>
    </li>
  </ul>
</div>
`;

exports[`components/Header/NavigationMobile when intelligentLink is false renders correctly 1`] = `
<div
  className="mobile-nav active"
>
  <div
    className="mobile-user-info"
  >
    <div
      className="Avatar Avatar-white Avatar-size-45"
    >
      <img
        alt="logo"
        src="https://appii.s3.eu-west-2.amazonaws.com/profile_images/1510235964102.jpg"
      />
    </div>
    <p>
      <strong>
        Jane
         
        Doe
      </strong>
      <a
        href="/profile"
        onClick={[Function]}
      >
        View Profile
      </a>
    </p>
  </div>
  <ul
    className="first"
  >
    <li>
      <a
        href="/personal-verification"
        onClick={[Function]}
      >
        My Verifications
      </a>
    </li>
    <li>
      <a
        className="h-verifications"
        href="/organisations/12/verifications"
        onClick={[Function]}
      >
        Applied Blockhain Ltd
      </a>
    </li>
    <li>
      <a
        href="/settings/account"
        onClick={[Function]}
      >
        Settings
      </a>
    </li>
    <li>
      <a
        href="https://appii.io/faqs/"
      >
        Help
      </a>
    </li>
  </ul>
  <ul>
    <li>
      Organisation Profile
    </li>
    <li>
      <a
        href="/organisations/12"
        onClick={[Function]}
      >
        Applied Blockhain Ltd
      </a>
    </li>
  </ul>
  <ul>
    <li>
      <a
        href="/people"
        onClick={[Function]}
      >
        People
      </a>
    </li>
    <li>
      <a
        href="/organisations"
        onClick={[Function]}
      >
        Organisations
      </a>
    </li>
  </ul>
  <ul>
    <li>
      <a
        href="/logout"
        onClick={[Function]}
      >
        Log Out
      </a>
    </li>
  </ul>
</div>
`;

exports[`components/Header/NavigationMobile when intelligentLink is valid renders correctly 1`] = `
<div
  className="mobile-nav active"
>
  <div
    className="mobile-user-info"
  >
    <div
      className="Avatar Avatar-white Avatar-size-45"
    >
      <img
        alt="logo"
        src="https://appii.s3.eu-west-2.amazonaws.com/profile_images/1510235964102.jpg"
      />
    </div>
    <p>
      <strong>
        Jane
         
        Doe
      </strong>
      <a
        href="/profile"
        onClick={[Function]}
      >
        View Profile
      </a>
    </p>
  </div>
  <ul
    className="first"
  >
    <li>
      <a
        href="/personal-verification"
        onClick={[Function]}
      >
        My Verifications
      </a>
    </li>
    <li>
      <a
        className="h-verifications"
        href="/organisations/12/verifications"
        onClick={[Function]}
      >
        Applied Blockhain Ltd
      </a>
    </li>
    <li>
      <a
        href="/settings/account"
        onClick={[Function]}
      >
        Settings
      </a>
    </li>
    <li>
      <a
        href="https://appii.io/faqs/"
      >
        Help
      </a>
    </li>
  </ul>
  <ul>
    <li>
      Organisation Profile
    </li>
    <li>
      <a
        href="/organisations/12"
        onClick={[Function]}
      >
        Applied Blockhain Ltd
      </a>
    </li>
  </ul>
  <ul>
    <li>
      <a
        href="/people"
        onClick={[Function]}
      >
        People
      </a>
    </li>
    <li>
      <a
        href="/organisations"
        onClick={[Function]}
      >
        Organisations
      </a>
    </li>
    <li
      className="hidden-xs"
    >
      <a
        href="http://google.co.uk"
      >
        CV Builder
      </a>
    </li>
  </ul>
  <ul>
    <li>
      <a
        href="/logout"
        onClick={[Function]}
      >
        Log Out
      </a>
    </li>
  </ul>
</div>
`;

exports[`components/Header/NavigationMobile when user has no organisations renders correctly 1`] = `
<div
  className="mobile-nav active"
>
  <div
    className="mobile-user-info"
  >
    <div
      className="Avatar Avatar-white Avatar-size-45"
    >
      <img
        alt="logo"
        src="https://appii.s3.eu-west-2.amazonaws.com/profile_images/1510235964102.jpg"
      />
    </div>
    <p>
      <strong>
        Jane
         
        Doe
      </strong>
      <a
        href="/profile"
        onClick={[Function]}
      >
        View Profile
      </a>
    </p>
  </div>
  <ul
    className="first"
  >
    <li>
      <a
        href="/personal-verification"
        onClick={[Function]}
      >
        My Verifications
      </a>
    </li>
    <li>
      <a
        href="/settings/account"
        onClick={[Function]}
      >
        Settings
      </a>
    </li>
    <li>
      <a
        href="https://appii.io/faqs/"
      >
        Help
      </a>
    </li>
  </ul>
  <ul>
    <li>
      <a
        href="/people"
        onClick={[Function]}
      >
        People
      </a>
    </li>
    <li>
      <a
        href="/organisations"
        onClick={[Function]}
      >
        Organisations
      </a>
    </li>
  </ul>
  <ul>
    <li>
      <a
        href="/logout"
        onClick={[Function]}
      >
        Log Out
      </a>
    </li>
  </ul>
</div>
`;
